### PR TITLE
Feature/DRP-10

### PR DESCRIPTION
**Pull request by ichvandi**
### PURPOSES :
[DRP-10](https://deuvox.atlassian.net/browse/DRP-10?atlOrigin=eyJpIjoiMzhhYzQ4ZTg0ZWUyNDJlOGEzMjk0YTQ4MWUxZGFjMTAiLCJwIjoiaiJ9)

### CHANGES :
- Add domain layer for Register
- Add data layer for register
- Add  utils for network request
- Add mapper
- Add delegate for viewmodel

### PR NOTES :
Please check for every Rx, is it already correct ?

### TEST RESULT :
Not yet 🤣

### CODE REVIEW

File: app/src/main/java/com/deuvox/deuvoxapp/feature/register/di/RegisterComponent.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.deuvox.deuvoxapp.feature.register.di
+
+import com.deuvox.deuvoxapp.base.anotation.scope.PerActivity
+import com.deuvox.deuvoxapp.di.component.AppComponent
+import dagger.Component
+
+@Component(dependencies = [AppComponent::class], modules = [RegisterModule::class])
+@PerActivity
+interface RegisterComponent {
+    // TODO: Inject Activity here!!!
```
Comment (ichsanachmad) [inline app/src/main/java/com/deuvox/deuvoxapp/feature/register/di/RegisterComponent.kt:10]: Change todo with this pattern 
//TODO (Ichsan): DRP-00X - Inject from UI

File: app/src/main/java/com/deuvox/deuvoxapp/feature/register/RegisterViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+) : BaseViewModel() {
+
+    init {
+        val param = RegisterParam(
+            "email@example.com",
+            "John Doe",
+            "jh0nD03",
+            "jh0nD03"
+        )
+        registerUseCase.buildUseCase(param)
```
Comment (ichsanachmad) [inline app/src/main/java/com/deuvox/deuvoxapp/feature/register/RegisterViewModel.kt]: please use execute(), i have been added execute method i think

**Review by ichsanachmad — APPROVED** [review]

**Comment by ichsanachmad** [comment]
Please add some test result (for logic i think we need to prove that data is showing maybe with screenshots of log/toast)

File: buildSrc/dependencies.gradle [hunk 3]
```diff
@@ -21,6 +21,7 @@ ext {
     junitVersion = '4.13.2'
     junitExtVersion = '1.1.3'
     espressoVersion = '3.4.0'
+    timberVersion = '5.0.1'
```
Comment (onirutlA) [inline buildSrc/dependencies.gradle:24]: Why add timber kak ?

**Review by Alturino — APPROVED** [review]

**Review by Farrize — APPROVED** [review]

**Pull request merged**